src/goto-analyzer/goto_analyzer_parse_options.cpp

```diff
@@ -200,6 +208,106 @@ void goto_analyzer_parse_optionst::get_command_line_options(optionst &options)
     options.set_option("text", true);
     options.set_option("outfile", "-");
   }
+
+  // The use should either select:
+  //  1. a specific analysis, or
+  //  2. a triple of task / analyzer / domain, or
+  //  3. one of the general display options
+
+  // Task options
+  if(cmdline.isset("show"))
+  {
+    options.set_option("show", true);
+    options.set_option("general-analysis", true);
+  }
+  else if(cmdline.isset("verify"))
+  {
+    options.set_option("verify", true);
+    options.set_option("general-analysis", true);
+  }
+  else if(cmdline.isset("simplify"))
+  {
+    options.set_option("simplify", true);
+    options.set_option("outfile", cmdline.get_value("simplify"));
+    options.set_option("general-analysis", true);
+
+    // For development allow slicing to be disabled in the simplify task
+    options.set_option(
+      "simplify-slicing",
+      !(cmdline.isset("no-simplify-slicing")));
+  }
+
+
+  if (options.get_bool_option("general-analysis"))
+  {
+    // Abstract interpreter choice
+    if(cmdline.isset("location-sensitive"))
+      options.set_option("location-sensitive", true);
+    else if(cmdline.isset("concurrent"))
+      options.set_option("concurrent", true);
+    else
+    {
+      // Silently default to location-sensitive as it's the "default"
+      // view of abstract interpretation.
+      options.set_option("location-sensitive", true);
+    }
+
+    // Domain choice
+    if(cmdline.isset("constants"))
+    {
+      options.set_option("constants", true);
+      options.set_option("domain set", true);
+    }
+    else if(cmdline.isset("dependence-graph"))
+    {
+      options.set_option("dependence-graph", true);
+      options.set_option("domain set", true);
+    }
+
+    if(!options.get_bool_option("domain set"))
```

Comment:
It's not quite an else because there are other places where the domain can be set.  For example, the handling of --show-intervals sets a domain and a task.